src/Evolution/DgSubcell/Tags/Jacobians.hpp

```diff
@@ -33,6 +35,124 @@ struct InverseJacobianLogicalToGrid : db::SimpleTag {
+                                             db::ComputeTag {
+  static constexpr size_t dim = Dim;
+  using base = InverseJacobianLogicalToGrid<Dim>;
+  using return_type = typename base::type;
+  using argument_tags = tmpl::list<
+      MapTagLogicalToGrid,
+      evolution::dg::subcell::Tags::Coordinates<dim, Frame::ElementLogical>>;
+  static void function(
+      const gsl::not_null<return_type*> inverse_jacobian_logical_to_grid,
+      const ElementMap<Dim, Frame::Grid>& logical_to_grid_map,
+      const tnsr::I<DataVector, dim, Frame::ElementLogical>& logical_coords) {
+    *inverse_jacobian_logical_to_grid =
+        logical_to_grid_map.inv_jacobian(logical_coords);
+  }
+};
+
+/// Compute item for the determinant of the inverse jacobian matrix
+/// from logical to grid coordinates
+template <size_t Dim>
+struct DetInverseJacobianLogicalToGridCompute
+    : DetInverseJacobianLogicalToGrid,
+      db::ComputeTag {
+  static constexpr size_t dim = Dim;
+  using base = DetInverseJacobianLogicalToGrid;
+  using return_type = typename base::type;
+  using argument_tags = tmpl::list<InverseJacobianLogicalToGrid<Dim>>;
+  static void function(
+      const gsl::not_null<return_type*> det_inverse_jacobian_logical_to_grid,
+      const ::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
+                              Frame::Grid>
+          inverse_jacobian_logical_to_grid) {
+    *det_inverse_jacobian_logical_to_grid =
+        determinant(inverse_jacobian_logical_to_grid);
+  }
+};
+
+/// Compute item for the inverse jacobian matrix from logical to
+/// inertial coordinates
+template <typename MapTagGridToInertial, size_t Dim>
+struct InverseJacobianLogicalToInertialCompute
+    : InverseJacobianLogicalToInertial<Dim>,
+      db::ComputeTag {
+  static constexpr size_t dim = Dim;
+  using base = InverseJacobianLogicalToInertial<Dim>;
+  using return_type = typename base::type;
+  using argument_tags =
+      tmpl::list<MapTagGridToInertial,
+                 evolution::dg::subcell::Tags::Coordinates<dim, Frame::Grid>,
+                 ::Tags::Time, ::domain::Tags::FunctionsOfTime,
+                 InverseJacobianLogicalToGrid<Dim>>;
+  static void function(
+      const gsl::not_null<return_type*> inverse_jacobian_logical_to_inertial,
+      const ::domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, dim>&
+          grid_to_inertial_map,
+      const tnsr::I<DataVector, dim, Frame::Grid>& grid_coords,
+      const double time,
+      const std::unordered_map<
+          std::string,
+          std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>&
+          functions_of_time,
+      const ::InverseJacobian<DataVector, Dim, Frame::ElementLogical,
+                              Frame::Grid>
+          inverse_jacobian_logical_to_grid) {
+    destructive_resize_components(inverse_jacobian_logical_to_inertial,
+                                  get<0>(grid_coords).size());
+    if (grid_to_inertial_map.is_identity()) {
```

Comment:
@nilsdeppe Can you check whether the if...else... statement here does what you want? I had to modify things a bit because your code snippets were assuming that we were using std::optional, while we decided earlier to no longer use optionals for the inverse jacobians.

Reply:
Yep, this is perfect!